File: Leetcode/arrays-and-strings/sliding-window/max-consecutive-ones-III/maxConsecutiveOnesIII.ts
function findMaxConsecutiveOnesIII(nums: number[], k: number): number {
    let left = 0, curr = 0, ans = 0;

    for (let right = 0; right < nums.length; right++) {
        if (nums[right] === 0) {
            curr++;
        }

        while (curr > k) {
            if (nums[left] === 0) {
                curr -= 1;
            }

            left++;
        }

        ans = Math.max(ans, right - left + 1);
    }

    return ans;
}

// Time: O(n);
// Space: O(n);

const result = findMaxConsecutiveOnesIII([1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0], 2);
console.log(result);

File: Leetcode/arrays-and-strings/sliding-window/max-consecutive-ones-III/maxConsecutiveOnesIII.java
public int maxConsecutiveOnes(int[] nums, int k) {
    int left = 0;
    int current = 0;
    int answer = 0;

    for (int right = 0; right < nums.length; right++) {
        if (nums[right] == 0) {
            current++;
        }

        while (current > k) {
            if (nums[left] == 0) {
                current--;
            }

            left++;
        }

        answer = Math.max(answer, right - left + 1);
    }

    return answer;
}
